third_party/2and3/jinja2/utils.pyi

```diff
@@ -1,26 +1,51 @@
-from typing import Any, Callable, Iterable, Optional
+import sys
+from typing import Any, Callable, IO, Iterable, Optional, Text, TypeVar, Union
 
 from markupsafe import Markup as Markup, escape as escape, soft_unicode as soft_unicode
 
 missing: Any
 internal_code: Any
 concat: Any
 
-def contextfunction(f): ...
-def evalcontextfunction(f): ...
-def environmentfunction(f): ...
-def internalcode(f): ...
-def is_undefined(obj): ...
+if sys.version_info >= (3, 6):
+    from builtins import _PathLike
+    _PathType = Union[bytes, Text, _PathLike]
+else:
+    _PathType = Union[bytes, Text]
+if sys.version_info >= (3,):
+    text_type = str
+else:
+    text_type = unicode
+
+_T = TypeVar('_T')
+
+class ContextFunction():
+    contextfunction: bool
+    def __call__(self, *args: Any, **kwargs: Any) -> _T: ...
+
+class EvalContextFunction():
+    evalcontextfunction: bool
+    def __call__(self, *args: Any, **kwargs: Any) -> _T: ...
+
+class EnvironmentFunction():
+    environmentfunction: bool
+    def __call__(self, *args: Any, **kwargs: Any) -> _T: ...
+
+def contextfunction(f: Callable[..., _T]) -> ContextFunction: ...
+def evalcontextfunction(f: Callable[..., _T]) -> EvalContextFunction: ...
+def environmentfunction(f: Callable[..., _T]) -> EnvironmentFunction: ...
+def internalcode(f: Callable[..., _T]) -> Callable[..., _T]: ...
```

Comment:
Should just use the `_CallableT` typevar I mentioned above, so the argument types are preserved.